src/Microsoft.PowerShell.CoreCLR.Eventing/DotNetCode/Eventing/EventProvider.cs

```diff
@@ -488,9 +488,9 @@ public bool WriteMessageEvent(string eventMessage)
         /// </param>
         /// <param name="eventPayload">
         /// </param>
-        public bool WriteEvent(ref EventDescriptor eventDescriptor, params object[] eventPayload)
+        public bool WriteEvent(in EventDescriptor eventDescriptor, params object[] eventPayload)
```

Comment:
Changing the signature on a public method like this is a breaking change

Reply:
The compiler doesn't permit overloading on ref/out/in differences, so this change is non-breaking.

Reply:
Technically the issue isn't with the compiler, but with the runtime. Modules compiled against the old API surface must continue to work after any changes are made to the API.

But I tested it out with a toy project (an exe and a lib, where I built an `in` version of the lib and then changed it so the exe is compiled against a `ref` version) and it seems to work.

Reply:
[The in, ref, and out keywords are not considered part of the method signature for the purpose of overload resolution. Therefore, methods cannot be overloaded if the only difference is that one method takes a ref or in argument and the other takes an out argument. ](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-parameter-modifier#:~:text=The%20in%2C%20ref%2C%20and%20out%20keywords%20are%20not,The%20following%20code%2C%20for%20example%2C%20will%20not%20compile%3A)

Reply:
Yeah that's still C# compiler rules, where the compiler statically decides which overload to resolve for a call. My concern was that the resolved method call in the IL might not match what the library was originally compiled against when its loaded at runtime (so even though the original program isn't broken it still requires recompilation, which is a dealbreaker for modules). It makes sense that that's not the case though.